File: cplus/leetcode/202606/algo3523.cpp
#include <bits/stdc++.h>
using namespace std;

// 贪心 八、其他
class Solution {
public:
    int maximumPossibleSize(vector<int>& nums) {
        int last = nums[0], ans = 1, n = nums.size();
        for (int i=1; i<n; i++) {
            if (nums[i] >= last) {
                last = nums[i]; ans += 1;
            }
        }
        return ans;
    }
};

File: python/algo/202606/algo3523.py
from typing import List

# 贪心 八、其他
class Solution:
    def maximumPossibleSize(self, nums: List[int]) -> int:
        last, ans = nums[0], 1
        for v in nums[1:]:
            if v >= last:
                last = v
                ans += 1
        return ans